### PR TITLE
aws-iot-device-client 1.9.2 integration (fix for secure tunneling with HTTP proxy)

### PR DESCRIPTION
**Pull request by brnaba-aws**
By submitting this pull request, I confirm that you can use, modify, copy, and
redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/secure_tunnel_watcher.py [hunk 1]
```diff
@@ -16,6 +16,32 @@
...
 LOCK_FILE_PATH = "/app/lock/"
 
+def extract_http_proxy_url(url):
+    # Splitting URL into different parts
+    parts = url.split('@')
+    # Extracting user, password, host, and port
+    if len(parts) == 2:  # Userinfo is present
+        userinfo, host_port = parts
+        user_pass = userinfo.split(':')
+        if len(user_pass) == 2:  # Both user and password are present
```
Comment (Kriechi) [inline src/secure_tunnel_watcher.py:26]: `2:` is a typo?

File: src/secure_tunnel_watcher.py [hunk 2]
```diff
@@ -16,6 +16,32 @@
...
+        host_port = parts[0]
+
+    host_port_split = host_port.split(':')
+    host = host_port_split[0]
+    if len(host_port_split) == 2:
+        port = host_port_split[1]
+    else:
+        port = None
+
+    return user, password, host, port
```
Comment (Kriechi) [inline src/secure_tunnel_watcher.py:43]: Please add tests to verify the different parsing behaviours with/without user-password-host-port combinations.

File: src/secure_tunnel_watcher.py [hunk 3]
```diff
@@ -53,6 +79,44 @@ def on_stream_event(self, event: IoTCoreMessage) -> None:
...
+                    print('Not able to parse proxy config: {}'.format(http_proxy_env))
+                    print('error: {}'.format(e))
+                    http_proxy_content["http-proxy-enabled"] = False
+                    
+            else:
+                http_proxy_content["http-proxy-enabled"] = False
+                    
+            with open(http_proxy_config, "w") as f:
+                    f.write(json.dumps(http_proxy_content))
+
```
Comment (Kriechi) [inline src/secure_tunnel_watcher.py:119]: Please extract all this into a function, and then add tests accordingly.

File: src/secure_tunnel_watcher.py [hunk 4]
```diff
@@ -53,6 +79,44 @@ def on_stream_event(self, event: IoTCoreMessage) -> None:
...
+                except Exception as e:
+                    print('Not able to parse proxy config: {}'.format(http_proxy_env))
+                    print('error: {}'.format(e))
+                    http_proxy_content["http-proxy-enabled"] = False
+                    
+            else:
+                http_proxy_content["http-proxy-enabled"] = False
+                    
+            with open(http_proxy_config, "w") as f:
+                    f.write(json.dumps(http_proxy_content))
```
Comment (Kriechi) [inline src/secure_tunnel_watcher.py:118]: You can directly `json.dump()` to the `f` file object without `write`.

File: recipe.yaml [hunk 5]
```diff
@@ -38,6 +38,7 @@ Manifests:
                 -e AWS_IOT_THING_NAME \
                 -e AWS_GG_NUCLEUS_DOMAIN_SOCKET_FILEPATH_FOR_COMPONENT \
                 -e SVCUID \
+                -e HTTP_PROXY \
```
Comment (Kriechi) [inline recipe.yaml:41]: Please add a sentence to the README to document the HTTP Proxy behaviour and option.

File: src/secure_tunnel_watcher.py [hunk 6]
```diff
@@ -16,6 +16,32 @@
 
 LOCK_FILE_PATH = "/app/lock/"
 
+def extract_http_proxy_url(url):
```
Comment (Kriechi) [inline src/secure_tunnel_watcher.py:19]: How is this different to the Python `urllib.parse(...)` function?https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse

File: recipe.yaml (hunk 5 above)
Comment (brnaba-aws) [reply recipe.yaml:41]: Ok. The HTTP_PROXY environment variable is automatically set by greengrass whenever a proxy is configured in the nucleus component configuration.

File: src/secure_tunnel_watcher.py (hunk 4 above)
Comment (brnaba-aws) [reply src/secure_tunnel_watcher.py]: done

**Comment by brnaba-aws** [comment]
@Kriechi , can you review the updated code? Thanks.

**Pull request merged**